api-gateway/src/main/java/com/petclinic/bffapigateway/config/GlobalServletExceptionHandler.java

```diff
@@ -0,0 +1,61 @@
+package com.petclinic.bffapigateway.config;
+
+
+import com.petclinic.bffapigateway.exceptions.*;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.web.reactive.error.ErrorWebExceptionHandler;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import org.springframework.web.server.ServerWebExchange;
+import reactor.core.publisher.Mono;
+
+import java.security.GeneralSecurityException;
+
+@Slf4j
+@Component
+public class GlobalServletExceptionHandler implements ErrorWebExceptionHandler {
+
+    @SuppressWarnings("NullableProblems")
+    @Override
+    public Mono<Void> handle(ServerWebExchange exchange, Throwable ex) {
+        HttpStatus status = HttpStatus.UNPROCESSABLE_ENTITY;
+
```

Comment:
Is it a good idea to set all your unknown exceptions that have not been specified to HttpStatus.UNPROCESSABLE_ENTITY ? Well, maybe it might not matter in the end since more exceptions will be added.